src/libcore/macros/mod.rs

```diff
@@ -556,13 +556,19 @@ macro_rules! unreachable {
     });
 }
 
-/// Indicates unfinished code by panicking with a message of "not yet implemented".
+/// Indicates unimplemented code by panicking with a message of "not implemented".
 ///
 /// This allows the your code to type-check, which is useful if you are prototyping or
 /// implementing a trait that requires multiple methods which you don't plan of using all of.
 ///
-/// There is no difference between `unimplemented!` and `todo!` apart from the
-/// name.
+/// The difference between `unimplemented!` and [`todo!`](macro.todo.html) is that while `todo!`
+/// conveys an intent of implementing the functionality later and the message is "not yet
+/// implemented", `unimplemented!` makes no such claims. Its message is "not implemented".
+/// Also some IDEs will mark `todo!`s.
+///
+/// There will be cases where you need to partially implement a trait or implement a function
+/// for only some variants of an enum. In those cases, a plain [panic!](macro.panic.html) with a
+/// suitable message is advisable.
```

Comment:
I don't find this paragraph helpful, recommend removing it. Implementing some trait methods but not others, or implementing for some enum variants but not others, is exactly what most people would consider "not implemented".

If you want to draw a distinction against panic (which I do not think is necessary to do, and based on https://github.com/rust-lang/rust/pull/67445#issuecomment-568314669 neither does withoutboats) then it should be the distinction I voiced in https://github.com/rust-lang/rust/pull/56348#issuecomment-447684679: if there is a contract that the caller violated in calling the not-implemented code, then the salient point is that the caller violated the contract, not that the code is not implemented.

Reply:
I've removed the paragraph. I'll leave differentiating from `panic!` to a followup PR, if any.